From 8c7b96f0d196259ac73614ec2483921a224d4229 Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Mon, 3 May 2010 11:14:47 +0800
Subject: [EVAL] Fix command -- crash

parse_command_args() returning a **argv pointer with *argv == 0 makes
dash segfault in find_command().  To reproduce run

 dash -c 'command --'

With this commit, parse_command_args() returns 0 if *argv is null after
parsing --, and so fixes the subsequent segfault.

Reported by Jonny through http://bugs.debian.org/579543

Signed-off-by: Gerrit Pape <pape@smarden.org>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 18071c7b6a847a7d8703c261eea522d5398fee90)

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 ChangeLog |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index cc47e28..57f2661 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2010-05-03  Gerrit Pape <pape@smarden.org>
+
+	* Fix command -- crash.
+
 2010-04-15  H. Peter Anvin <hpa@zytor.com>
 
 	* Fix for job control off warning.
-- 
1.7.4.1

